packages/@react-types/shared/src/dnd.d.ts

```diff
@@ -142,7 +176,31 @@ export interface DroppableCollectionProps {
   /** Handler that is called when a valid drag exits the drop target. */
   onDropExit?: (e: DroppableCollectionExitEvent) => void,
   /** Handler that is called when a valid drag is dropped on the drop target. */
-  onDrop?: (e: DroppableCollectionDropEvent) => void
+  onDrop?: (e: DroppableCollectionDropEvent) => void,
+  /**
+   * Handler called when external or copied items are dropped "between" the droppable collection's items.
+   */
+  onInsert?: (e: DroppableCollectionInsertDropEvent) => void,
+  /**
+   * Handler called when external items are dropped on the droppable collection's root.
+   */
+  onRootDrop?: (e: DroppableCollectionRootDropEvent) => void,
+  /**
+   * Handler called when items are dropped "on" a droppable collection's item.
+   */
+  onItemDrop?: (e: DroppableCollectionOnItemDropEvent) => void,
+  /**
+   * Handler called when items are reordered via drag in the source collection.
+   */
+  onReorder?: (e: DroppableCollectionReorderEvent) => void,
+  /**
+   * The drag types that the droppable collection accepts.
+   */
+  acceptedDragTypes?: 'all' | Array<string>,
+  /**
+   * A function returning whether a given target in the droppable collection is a valid "on" drop target for the current drag types.
+   */
+  isValidDropTarget?: (target: ItemDropTarget, types: DragTypes) => boolean
```

Comment:
Think about name, make more specific to "on" drop targets

suggestions:
shouldAcceptItemDrop
brainstorm more